pkg/extension/sumologicextension/extension.go

```diff
@@ -286,10 +288,20 @@ func (se *SumologicExtension) registerCollector(ctx context.Context, collectorNa
 			zap.Int("status_code", res.StatusCode),
 			zap.String("response", buff.String()),
 		)
-		return CollectorCredentials{}, fmt.Errorf(
-			"failed to register the collector, got HTTP status code: %d",
-			res.StatusCode,
-		)
+
+		// Return unrecoverable error for 4xx status codes except 429
+		if res.StatusCode >= 400 && res.StatusCode < 500 && res.StatusCode != 429 {
```

Comment:
We might consider reusing this abstraction: https://github.com/open-telemetry/opentelemetry-collector/blob/1fada742275967c44c7c01445f097793b9d0caae/consumer/consumererror/permanent.go

Reply:
Could you check it now? as abstraction is not exported I use backoff.Pernament